### PR TITLE
Eledger 117 create exception handler for project

### LINKED ISSUE #34: Eledger-117 create exception handler classes and  use it in our application.

### PR DESCRIPTION
**Pull request by kharya1996**

### CODE REVIEW

File: src/main/java/in/pune/royforge/eledgerapi/data/dao/WalletDAOImpl.java [hunk 1]
```diff
@@ -82,21 +82,26 @@ private void updateWallet(WalletEntity walletEntity, WalletTransaction wallet) {
...
+	 */
 	@Override
 	public List<WalletData> getWallets() {
 		List<WalletData> wallets = new ArrayList<>();
 		Iterable<WalletEntity> walletsList = walletEntityRepository.findAll();
-		for (WalletEntity walletEntity : walletsList) {
-			WalletData walletData = new WalletData();
-			setWalletData(walletEntity, walletData);
-			wallets.add(walletData);
+		if (walletsList != null) {
```
Comment (abkesarwani) [inline src/main/java/in/pune/royforge/eledgerapi/data/dao/WalletDAOImpl.java]: use null check like (if ( null != walletsList ) in all places

File: src/main/java/in/pune/royforge/eledgerapi/data/service/TransactionServiceImpl.java [hunk 2]
```diff
@@ -8,41 +8,46 @@
 import org.springframework.stereotype.Service;
 import in.pune.royforge.eledgerapi.data.dao.ITransactionDAO;
 import in.pune.royforge.eledgerapi.data.model.Transaction;
+import in.pune.royforge.eledgerapi.exceptionhandler.RecordNotFoundException;
 
 @Service
 public class TransactionServiceImpl implements TransactionService {
+	public void save(Transaction transaction) {
```
Comment (abkesarwani) [inline src/main/java/in/pune/royforge/eledgerapi/data/service/TransactionServiceImpl.java]: I think we don't need this.

File: src/main/java/in/pune/royforge/eledgerapi/exceptionhandler/CustomExceptionHandler.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+	public final ResponseEntity<Object> handleRecordNotFoundException(RecordNotFoundException ex) {
+		Date currentDate = new Date();
+		ErrorResponse error = new ErrorResponse(currentDate, ex.getMessage(), HttpStatus.NOT_FOUND);
+		return new ResponseEntity<Object>(error, HttpStatus.NOT_FOUND);
+	}
+
+
+	@ExceptionHandler(EmptyListException.class)
+	public final ResponseEntity<Object> handleEmptyListException(EmptyListException ex) {
+		Date currentDate = new Date();
```
Comment (abkesarwani) [inline src/main/java/in/pune/royforge/eledgerapi/exceptionhandler/CustomExceptionHandler.java]: you may use directly  in ResponseEntity object.

**Review by mahek-omshree — APPROVED** [review]
Custom exception is handled properly.

**Pull request merged**